public/tabs/__tests__/chat_window_header.test.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render, screen, fireEvent, act } from '@testing-library/react';
+
+import { ChatWindowHeader, ChatWindowHeaderProps } from '../chat_window_header';
+import * as chatContextExports from '../../contexts/chat_context';
+import { TabId } from '../../types';
+
+jest.mock('../../components/chat_window_header_title', () => {
+  return { ChatWindowHeaderTitle: () => <div>OpenSearch Assistant</div> };
+});
+
+const setup = ({
+  selectedTabId,
+  ...props
+}: Partial<ChatWindowHeaderProps> & { selectedTabId?: TabId } = {}) => {
+  const useChatContextMock = {
+    sessionId: '1',
+    title: 'foo',
+    selectedTabId: selectedTabId || 'chat',
+    setSessionId: jest.fn(),
+    setTitle: jest.fn(),
+    setFlyoutVisible: jest.fn(),
+    setSelectedTabId: jest.fn(),
+    setFlyoutComponent: jest.fn(),
+  };
+  jest.spyOn(chatContextExports, 'useChatContext').mockReturnValue(useChatContextMock);
```

Comment:
I am thinking if we should add two files: chat_context.mock.ts and use_chat_state.mock.ts, in which we can mock the output and export the mocked methods. I've seen over 3 PRs have this spyOn function to hijack useChatContext.

Reply:
I think we can provide a general mocks for `chat_context` and `use_chat_state`. But we may still need to customized the mock in some specific PRs. Can be refactor in the next PRs.